### DISCUSSION TITLE
What is the purpose of indicating whether a codepoint is whitespace or not when advancing in an external scanner?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1255] [Q&A] [answered]
From https://tree-sitter.github.io/tree-sitter/creating-parsers#external-scanners:
> `void (*advance)(TSLexer *, bool skip)` - A function for advancing to the next character. If you pass `true` for the second argument, the current character will be treated as whitespace.

I've been doing this in my external scanner, but I don't know what functionality it actually facilitates. Some tokens (for example, quoted strings) could contain whitespace in the sense of space codepoints, but that is probably different than the whitespace that tree-sitter means here, right?

**Comment by maxbrunsfeld** [comment, marked as answer]
It influences the start position of the token that you create in the syntax tree. Leading white space is not considered part of the token itself.

**Reply by ahelwer** [reply]
Got it, so if I actually want leading whitespace to be included in the token I should mark it as not-whitespace.
